src/plots/plot.ts

```diff
@@ -451,27 +456,42 @@ export class Plot extends Component {
    * @return {Plots.PlotEntity[]}
    */
   public entities(datasets = this.datasets()): Plots.PlotEntity[] {
-    var entities: Plots.PlotEntity[] = [];
+    var entities = this._lightweightEntities(datasets);
+
+    entities.forEach((entity) => {
+      entity.selection = entity.drawer.selectionForIndex(entity.validDatumIndex);
+      delete(entity.drawer);
+      delete(entity.validDatumIndex);
```

Comment:
`map()` is harder to implement because we don't know what are the properties on the `entity`. If we hardcode them, we have to maintain this bit of code, which leads to fragility